app/forms/mappings/LocalDateFormatter.scala

```diff
@@ -72,16 +57,41 @@ private[mappings] class LocalDateFormatter(
       .toList
 
     fields.count(_._2.isDefined) match {
-      case 3 =>
-        formatDate(key, data).left.map {
-          _.map(_.copy(key = key, args = args))
+      case 3 => Nil
+      case 2 => Seq(FormError(key, oneRequiredKey, missingFields ++ args))
+      case 1 => Seq(FormError(key, twoRequiredKey, missingFields ++ args))
+      case _ => Seq(FormError(key, allRequiredKey, args))
+    }
+  }
+
+  private def handleRangeInputs(key: String, data: Map[String, String]): Seq[FormError] = {
+    def isValid(value: Option[String], range: Range): Boolean =
+      value.flatMap(v => Try(v.trim.toInt).toOption).exists(range.contains)
+
+    val dayValid = isValid(data.get(s"$key.day"), 1 to 31)
+    val monthValid = isValid(data.get(s"$key.month"), 1 to 12)
+    val yearValid = isValid(data.get(s"$key.year"), 1 to Int.MaxValue)
```

Comment:
This check might be better to be `.length() == 4` so that if someone accidentally puts a lower or higher year we catch it. I've noticed I've managed to put 20254 before by accident.

Reply:
or, if you want to keep it in the same format. Then something like 1900 to 9999 🤷‍♂️